tiledb/sm/array_schema/domain.cc

```diff
@@ -272,7 +272,7 @@ int Domain::cell_order_cmp_2(const void* coord_a, const void* coord_b) {
 
 int Domain::cell_order_cmp(
     const type::DomainDataRef& left, const type::DomainDataRef& right) const {
-  if (cell_order_ == Layout::ROW_MAJOR) {
+  if (cell_order_ == Layout::ROW_MAJOR || cell_order_ == Layout::HILBERT) {
```

Comment:
Is this the only case in this file that `Layout::HILBERT` should be treated as `Layout::ROW_MAJOR`? I see [get_end_of_cell_slab](https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/array_schema/domain.cc#L430) and [stride](https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/array_schema/domain.cc#L581) have a similar Layout checks assuming only `ROW_MAJOR` and `COL_MAJOR`, should `HILBERT` be added there explicitly to avoid other hidden bugs?

Reply:
https://github.com/TileDB-Inc/TileDB/pull/4926 for [get_end_of_cell_slab](https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/array_schema/domain.cc#L430).

[stride](https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/array_schema/domain.cc#L581) is a dense only function and Hilbert is not supported in dense.

Reply:
Then NIT: Let's make code self-documented and adapt `stride` to throw with proper message if Layout is not ROW or COL MAJOR.